### PR TITLE
Add ssh-persistent-sessions.md to document how to achieve a persistent session with ssh

### PR DESCRIPTION
**Pull request by traversaro**
Based on original (non-public) docs from @diegoferigo

### CODE REVIEW

**Review by S-Dafarra — APPROVED** [review]

File: doc/ssh-persistent-sessions.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+### Tricks
+
+#### Copy text from a tmux session
+
+Tmux has a very particular way to handle splits (terminals part of the view). Mouse, keyboard, and scrollback buffers are treated in a way that could make most of new users puzzled.
+
+Pasting content from the clipboard of the local machine should work OOTB, though copying text from the remote machine does not work.
+
+The trick is to press `Shift` while using the mouse to select the text to copy. This operation disables the mouse supports and gives back copy capability.
```
Comment (flferretti) [inline doc/ssh-persistent-sessions.md:27]:
```suggestion
### Setup
1. First enable mouse support `echo "set -g mouse on" >> $HOME/.tmux.conf`
1. Then, while in a tmux session press `Shift` while using the mouse to select the text to copy. This operation disables the mouse supports and gives back copy capability.

Alternatively, you can use the `tmux` command `copy-mode`: `Ctrl+b` `Ctrl+[` to enter copy mode, then use the arrow keys to move around and select text. Press `Enter` to copy the selected text to the clipboard.

### Setup
1. First enable mouse support `echo "set -g mouse on" >> $HOME/.tmux.conf`
1. Then, while in a tmux session press `Shift` while using the mouse to select the text to copy. This operation disables the mouse supports and gives back copy capability.

Alternatively, you can use the `tmux` command `copy-mode`: `Ctrl+b` `Ctrl+[` to enter copy mode, then use the arrow keys to move around and select text. Press `Enter` to copy the selected text to the clipboard.

### Usage

```


Can we make this more explicit, so it easier to find for new users? 
It doesn't allow me to put one more code snippet after `### Usage` in suggestion so I'll paste it below:

```bash
# Check existing sessions
tmux ls

# Create a new named session
tmux new -s <name>

# Detach from a session
# Press Ctrl+b d

# List existing sessions
tmux ls

# Delete a session
tmux kill-session -t <name>

# Reconnect to an existing session
tmux attach -t <name>
```

File: doc/ssh-persistent-sessions.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+## byobu
+
+For an exhaustive tutorial, refer to the [official documentation](https://www.byobu.org/documentation).
```
Comment (flferretti) [inline doc/ssh-persistent-sessions.md:31]:
```suggestion
Optionally, you can use `byobu` to wrap `tmux` and `screen` and provide a more user-friendly interface.

For an exhaustive tutorial, refer to the [official documentation](https://www.byobu.org/documentation).
```
As Byobu is not strictly necessary for the main objective of achieving a persistent session in SSH, can we make it more clear that it is as an optional requirement?

**Review by xela-95 — APPROVED** [review]

File: doc/ssh-persistent-sessions.md (hunk 1 above)
Comment (traversaro) [reply doc/ssh-persistent-sessions.md]: Done : https://github.com/ami-iit/ami-commons/commit/aa4f155d6dae74260890325552c5cfb288d802be .

File: doc/ssh-persistent-sessions.md [hunk 3]
```diff
@@ -28,16 +28,26 @@ Pasting content from the clipboard of the local machine should work OOTB, though
...
+
+# Detach from a session
+# Press Ctrl+B D
+
+# Delete a session
+tmux kill-session -t <name>
+
+# Reconnect to an existing session
+tmux attach -t <name>
+~~~
```
Comment (flferretti) [inline doc/ssh-persistent-sessions.md:50]: Probably I'm missing something, is there a reason for the `~~~` instead of the code snippet?

Comment (traversaro) [reply doc/ssh-persistent-sessions.md:50]: The other is copy pasted. Let me harmonize the file.

**Review by flferretti — APPROVED** [review]

File: doc/ssh-persistent-sessions.md (hunk 3 above)
Comment (flferretti) [reply doc/ssh-persistent-sessions.md]: Okay, thanks a lot

**Pull request merged**